packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -339,15 +339,25 @@ export async function ensureAvailableUsers(
     });
   }
 
+  const getStartTime = (startTimeInput: string, timeZone?: string) => {
```

Comment:
This code was taken from `packages/trpc/server/routers/viewers/slots/util.ts` so that we mirror how the `getUserAvailability` function is called.

I noticed that the `.utc()` function was being call for the `endTime` but not the `startTime`. Have added that here and will look to see if `util.ts` needs it to be added as well.

Reply:
Is the startTimeInput in Zulu format or something like `2024-10-1T05:00` ?

Reply:
If it isn't in Zulu format or ISO format. It reads the time in local tz. So if you run `dayjs("2024-10-01T05:00")` in India.

1. It will be `2024-10-01T05:00:00+05:30`. 
2. Then we convert it into the specified timezone.
3. And then to UTC. 

Which I guess isn't what we want.

I think here we are thinking of reading `2024-10-01T05:00` in the specified timezone.
For that you need to do something like `dayjs.tz(2024-10-01T05:00, timeZone)`

Reply:
And I guess github actions are run in UTC tz, so It won't show any errors. And the productions server will be fine to as long as it is in UTC too.

Reply:
Yeah this is exactly the problem. This function keeps the date in the time zone specified by the input but along the path of checking availability, it's never converted to UTC so time slots don't line up, resulting in "No available users" errors

Reply:
It's in ISO format with the timezone information included. So, for example, if on the booking page you select New York for the time zone, it will pass `2024-02-17T12:00:00-05:00`.

Reply:
The main problem is that we were calling `stringToDayjs` with this string and it was never been converted to UTC before doing all the appropriate checks and so the time slots weren't lining up properly, resulting in "No available users found" errors.